### DISCUSSION TITLE
What linting configuration should I use for my Agoric contracts?

### DISCUSSION THREAD

**Discussion by rowgraus** [Agoric/agoric-sdk#8274] [Q&A]
context:

 - https://github.com/Agoric/agoric-sdk/discussions/8280

_p.s. edited to separate question from answer_

**Comment by dckc** [comment]
Agoric recommends the following configuration:

Linting Configuration
Adapted from [Jessie ESLint Configuration](https://docs.agoric.com/guides/js-programming/hardened-js.html#tool-support-eslint-config)

1. Run:
```
yarn add -D \
 eslint \
 @agoric/eslint-config@dev \
 @endo/eslint-plugin \
 @jessie.js/eslint-plugin \
  eslint-config-airbnb-base \
 eslint-plugin-jsdoc \
 eslint-config-prettier \
 eslint-plugin-import \
 eslint-plugin-github
```

2. Add the following to your `package.json`
```
"eslintConfig" : {
  "parserOptions": {
      "sourceType": "module",
      "ecmaVersion": 6
  },
  "extends": [
    "@agoric"
  ]
}
```
3. Run `yarn eslint '**/*.js'` (or specify other files/directory globs)

**Reply by 0xpatrickdev** [reply]
I think this may need to be `"ecmaVersion": 12` or `"ecmaVersion": 2021` to support things like `bigint`

**Comment by dckc** [comment]
@turadg does the advice above still look good?
_no rush_

**Reply by turadg** [reply]
incomplete. I'll try to find time to flesh it out. https://github.com/CrabblePitch/crabbleProtocol/pull/124 provides some direction

**Reply by dckc** [reply]
I see https://github.com/Agoric/dapp-offer-up/pull/26#discussion_r1440876220 as well

**Comment by dckc** [comment]
@erights asked in https://github.com/Agoric/dapp-agoric-basics/pull/29#discussion_r1538587410
> I `yarn format` in this repo?

Suggestions, anyone?

**Reply by erights** [reply]
Answered by @turadg at https://github.com/Agoric/dapp-agoric-basics/pull/29#discussion_r1540165688

**Comment by dckc** [comment]
`@import` is another recent relevant development:

 - https://github.com/Agoric/agoric-sdk/pull/9149

**Comment by dckc** [comment]
I'm struggling to configure lint in another project. I think I'll promote the symptoms to their own question:

 - https://github.com/Agoric/agoric-sdk/discussions/9892
